libs/langchain/langchain/retrievers/self_query/elasticsearch.py

```diff
@@ -61,11 +61,12 @@ def visit_comparison(self, comparison: Comparison) -> Dict:
         ]
 
         if is_range_comparator:
-            return {
-                "range": {
-                    field: {self._format_func(comparison.comparator): comparison.value}
+            value = comparison.value
+            if isinstance(comparison.value, dict) and "date" in comparison.value:
+                value = comparison.value["date"]
+                return {
+                    "range": {field: {self._format_func(comparison.comparator): value}}
```

Comment:
```suggestion
            return {
                "range": {field: {self._format_func(comparison.comparator): value}}
```
I think indentation got mixed up here